noodles-editor/src/noodles/components/block-library-ranking.ts

```diff
@@ -0,0 +1,128 @@
+import type { NodeType } from '../utils/node-creation-utils'
+import { getNodeDescription, typeCategory, typeDisplayName } from './op-components'
+
+// Common operators that should be boosted in search results.
+// These are operators that users frequently need and should appear higher in results.
+const COMMON_OPERATORS_BOOST: Record<string, number> = {
+  FileOp: 100,
+  NumberOp: 100,
+  MathOp: 100,
+  ColorOp: 100,
+  GeoJsonOp: 100,
+  MaplibreBaseMapOp: 80,
+  ScatterplotLayerOp: 80,
+  ArcLayerOp: 80,
+  GeoJsonLayerOp: 80,
+  HexagonLayerOp: 80,
+}
+
+// Popular operators shown when search is empty.
+// Ordered by importance/frequency of use.
+export const POPULAR_OPERATORS: NodeType[] = [
+  'FileOp',
```

Comment:
These are separate lists and kept up to date manually. We could do something more advanced in the future but this is simple enough